helix-term/src/application.rs

```diff
@@ -73,6 +76,8 @@ impl Application {
             editor,
 
             callbacks: FuturesUnordered::new(),
+
+            lsp_progress: HashMap::new(),
```

Comment:
Having this on `Application` is problematic because two different LSPs could write to the same progress token. I know this is kind of hard, but we need to set these on the LSP client :/